core/src/main/java/org/springframework/security/authentication/UsernamePasswordAuthenticationToken.java

```diff
@@ -71,6 +75,11 @@ public UsernamePasswordAuthenticationToken(Object principal, Object credentials,
 		super.setAuthenticated(true); // must use super, as we override
 	}
 
+	public static UsernamePasswordAuthenticationToken authenticated(Object principal, Object credentials,
+			Collection<? extends GrantedAuthority> authorities) {
```

Comment:
It would be nice if Spring Security started using these methods. Can you search for places in the codebase that use the constructors and change them to use the static methods?